hack/e2e/run-rp-and-e2e.sh

```diff
@@ -65,9 +65,31 @@ register_sub() {
 }
 
 run_e2e() {
-    RESOURCEGROUP=$ARO_RESOURCEGROUP make test-e2e
+    export E2E_ENTRYPOINT="docker run --privileged \
+                --network host \
+                --env-file <(env | egrep 'AZURE|E2E|ARO') \
+                --env LOCATION \
+                --env CI \
+                --env RESOURCEGROUP=$ARO_RESOURCEGROUP \
+                --env CLUSTER=$CLUSTER \
+                --env RP_MODE \
+                --entrypoint e2e.test $ARO_IMAGE -test.timeout 180m -test.v -ginkgo.v"
+    RESOURCEGROUP=$ARO_RESOURCEGROUP E2E_ENTRYPOINT=$E2E_ENTRYPOINT make test-e2e
```

Comment:
I think I'm +1 on not duplicating `-test.timeout 180m -test.v -ginkgo.v` and keeping the canonical version in the Makefile.  What I don't like here is that currently this call will still make an (unused) local e2e.test and hence require the Go compiler toolchain, which I want to get rid of in the nightly e2e path.
Perhaps changing the Makefile something like the following will help?
```
TEST_E2E ?= go test ./test/e2e -tags e2e

test-e2e:
	${TEST_E2E} -test.timeout 180m -test.v -ginkgo.v
```